Sources/SourceKitLSP/Swift/SwiftTestingScanner.swift

```diff
@@ -95,7 +95,20 @@ struct TestingAttributeData {
           return false
         }
       }.flatMap(\.arguments)
-      .compactMap { $0.expression.as(StringLiteralExprSyntax.self)?.representedLiteralValue }
+      .compactMap {
+          if let stringLiteral = $0.expression.as(StringLiteralExprSyntax.self) {
+            return stringLiteral.representedLiteralValue
```

Comment:
When defining tags is it possible to have both `"foo"` and `.foo` on the same test? The `tags` trait won't let you mix and match the two, and so the first updated test in this PR actually won't compile (I'll fix that).

For example, this won't compile: `@Test(.tags("foo", .foo)) // Cannot convert value of type 'String' to expected argument type 'Tag'`

As a side effect of this I can't think of a way to run in to a conflict between `"foo"` and `.foo` both resolving to the string "foo" because you can't use them both in the same place.

Reply:
Yes, `"foo"` and `.foo` are in separate namespaces and can both be present on the same test:

```swift
@Test(.tags("foo"), .tags(.foo)) func f() {}
```

Reply:
But more importantly, if I ask to run all tests with tag `.foo`, I don't expect to run tests with tag `"foo"`. (Or, maybe I do expect it, but this is one of the hard subjective questions around tags and the answer we're mandating is that they are different tags and should not be confused by the tools.)

Reply:
Ah, I didn't think about adding multiple `.tags` traits to the test. 

This creates some tricky UX for tools, since the more type safe way to declare tags is obviously using the static `.foo` syntax, but the more intuitive way for users to use tags in tools is just `"foo"`, without regard for how they're declared. How tags are declared should be considered an implementation detail.

I also think requiring users to use the string ".foo" when tags are declared statically pushes them towards using string tags to avoid the annoyance, which isn't ideal when static tags are the safer choice.

Maybe in swift-testing `.foo` aliases `"foo"` unless there is a namespace conflict? Then swift-testing emits a warning? I'm not sure of a great solution here.

Reply:
They do not alias each other, no.

Reply:
Done